src/main/java/com/mparticle/kits/UrbanAirshipKit.java

```diff
@@ -303,28 +297,36 @@ private boolean logAirshipRetailEvents(CommerceEvent event) {
         switch (event.getProductAction()) {
             case Product.PURCHASE:
                 for (Product product : event.getProducts()) {
-                    populateRetailEvent(RetailEvent.createPurchasedEvent(), product).track();
+                    populateRetailEventTemplate(RetailEventTemplate.newPurchasedTemplate(), product)
+                            .createEvent()
+                            .track();
```

Comment:
@rlepinski this may be an issue with the current implementation as well. Does `track()` call the UA singleton under the hood? Or are these events being created and then not actually logged?